From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Nicolas Iooss <nicolas.iooss_linux@m4x.org>
Date: Mon, 20 Dec 2021 16:44:35 +0100
Subject: [PATCH] {BUG in 5.16-rc2} fs/sysv/itree: return res instead of blocks

Found with clang on x86-64:

    fs/sysv/itree.c:433:19: error: variable 'res' set but not used
    [-Werror,-Wunused-but-set-variable]
            unsigned blocks, res, direct = DIRECT, i = DEPTH;
                             ^
---
 fs/sysv/itree.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/fs/sysv/itree.c b/fs/sysv/itree.c
index d4ec9bb97de9..aa3d46d2375c 100644
--- a/fs/sysv/itree.c
+++ b/fs/sysv/itree.c
@@ -438,7 +438,7 @@ static unsigned sysv_nblocks(struct super_block *s, loff_t size)
 		res += blocks;
 		direct = 1;
 	}
-	return blocks;
+	return res /* BUG: blocks*/;
 }
 
 int sysv_getattr(struct user_namespace *mnt_userns, const struct path *path,
-- 
